src/server.js

```diff
@@ -0,0 +1,319 @@
+const WINDOW_SIZE = process.env.WINDOW_SIZE || "800,600";
+const WINDOW_POSITION = process.env.WINDOW_POSITION || "0,0";
+const PERSISTENT_DATA = process.env.PERSISTENT || '0';
+const REMOTE_DEBUG_PORT = process.env.REMOTE_DEBUG_PORT || 35173;
+const FLAGS = process.env.FLAGS || null;
+
+// Environment variables which can be overriden from the API
+var kioskMode = process.env.KIOSK || '0';
+var enableGpu = process.env.ENABLE_GPU || '0';
+
+var DEFAULT_FLAGS = [];
+var currentUrl = '';
+var flags = [];
+
+// Returns the URL to display, adhering to the hieracrchy:
+// 1) the configured LAUNCH_URL
+// 2) a discovered HTTP service on the device
+// 3) the default static HTML
+async function getUrlToDisplayAsync() {
+    var launchUrl = process.env.LAUNCH_URL || null;
+    if (null != launchUrl)
+    {
+      console.log("Using LAUNCH_URL: " + launchUrl)
+
+      // Prepend http:// if the LAUNCH_URL doesn't have it.
+      // This is needed for the --app flag to be used for kiosk mode
+      if (!/^https?:\/\//i.test(launchUrl)) {
+        launchUrl = 'http://' + launchUrl;
+      }
+
+      return launchUrl;
+    }
+
+    console.log("LAUNCH_URL environment variable not set.")
+    console.log("Looking for local HTTP/S services.")
+
+    // Check each HTTP/S port
+    var ports = [80,443,8080];
+    var returnURL = null;
+    await Promise.all(ports.map(async (port) => 
+    {
+      var url = '';
+      //prepend the correct prefix
+      if(port == 443)
+      {
+        url = 'https://'
+      }
+      else
+      {
+        url = 'http://'
+      }
+      url = url + 'localhost:' + port;
+      
+      console.log("Trying " + url);
+
+      try 
+      {
+        //request the URL
+        const request = bent(url);
+        const response = await request();
+        //If OK
+        if (200 == response.statusCode)
+        {
+          console.log("HTTP/S services found at: " + url)
+          return url;
+        }
+      }
+      catch(error)
+      {
+        //TODO: Handle expected connection errors, but re-throw anything else
+      }
+    })).then((urls) =>
+    {
+      // An array of the promise resolves will be returned. The discovered services
+      // will have their URL in the array. The others will be null.
+      var filteredUrls = urls.filter(function (el) {
+        // Only keep the non-null items
+        return el != null;
+      });
+
+      // If we found a URL
+      if(filteredUrls.length > 0)
+      {
+        returnURL = filteredUrls[0];
+      }
+      // Otherwise send the default HTML
+      else
+      {
+        console.log("Displaying default HTML page");
+       returnURL = "file:///home/chromium/index.html";
+      }
+    });
+
+    return returnURL;
+  }
+       
+// Launch the browser with the URL specified
+let launchChromium = function(url) {
+  chromeLauncher.killAll().then(() => { 
+
+    flags = [];
+    // If the user has set the flags, use them
+    if (null != FLAGS)
```

Comment:
Not sure if there is any difference but I've almost never seen the variable on the right side of a comparison in js. It probably works exactly in the same way, so most likely a stylistic thing.

Reply:
@tmigone you're right - it's poor practice in any language, since accidentally typing '=' rather than '==' or '===' can create odd bugs which are hard to track down. It's my sloppiness coming out just because I'm hacking something in a language I don't fully get. :)